contracts/l2/dai.sol

```diff
@@ -96,17 +93,19 @@ contract Dai {
 
   // --- ERC20 Mutations ---
   function transfer(address to, uint256 value) external returns (bool) {
+    require(to != address(0), "Dai/invalid-address");
     uint256 balance = balanceOf[msg.sender];
     require(balance >= value, "Dai/insufficient-balance");
 
     balanceOf[msg.sender] = balance - value;
-    balanceOf[to] = add(balanceOf[to], value);
+    balanceOf[to] = balanceOf[to] + value;
 
     emit Transfer(msg.sender, to, value);
 
     return true;
   }
   function transferFrom(address from, address to, uint256 value) external returns (bool) {
+    require(to != address(0), "Dai/invalid-address");
```

Comment:
I would add another require for `from` as well.
`require(from != address(0), "Dai/invalid-address");`

Reply:
Setting `from` to `address(0)` will fail anyways as there is no allowance. We don't need to add this explicit check.